File: mario.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

void pyramid(int input_pyramid_height);
int main(void)
{
    int input_pyramid_height;
    do
    {
        //  printf("Please enter fitting integer: ");
         printf("Height: ");
         input_pyramid_height = get_int();
    }
    while(input_pyramid_height < 0 || input_pyramid_height > 23);
    if(input_pyramid_height >= 0 && input_pyramid_height < 24)
    {
        pyramid(input_pyramid_height);
    }
}
void pyramid(int input_pyramid_height)
{

    string spaces = " ";
    string hashes = "#";
    for(int i = 1; i <= input_pyramid_height; i++)
        {
            
            
            int x = input_pyramid_height-i-1; 
            for(int space_generator = x ; space_generator>=0;space_generator--)
            {
                printf("%s",spaces);
            }
            for(int hash_generator = input_pyramid_height; hash_generator > x ;  hash_generator--)
            {
                printf("%s",hashes);
            }
            
            printf("\n");
        }
    
    
}

File: mario.py
import cs50


def get_positive_int():
    while True:
        n = cs50.get_int()
        if n > 0:
            break
    return n

def main():
    while True:
        print("Height:",end = " ")
        input_pyramid_height = get_positive_int()
        if input_pyramid_height >= 0 and input_pyramid_height < 24:
            break
    print_pyramid(input_pyramid_height)

def print_pyramid(input_pyramid_height):
    for i in range(input_pyramid_height):
        space_retain = input_pyramid_height - i - 1
        for x in range(space_retain,0,-1):
            print(' ',end="")
        for y in range(input_pyramid_height, space_retain-1,-1):
            print('#',end="")
        print("")


if __name__ == "__main__":
    main()
